### PR TITLE
fix: TRLPlusCCancellationTokenSource doesn't cancel current TestRun

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/3402

### CODE REVIEW

File: src/Adapter/MSTest.TestAdapter/IPlatformServiceProvider.cs [hunk 1]
```diff
@@ -117,4 +117,6 @@ ITestSourceHost CreateTestSourceHost(
     /// This was required for compatibility reasons since the TestContext object that the V1 adapter had for desktop is not .Net Core compliant.
     /// </remarks>
     ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties);
+
+    TestRunCancellationToken? Token { get; set; }
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/IPlatformServiceProvider.cs]: Let's not change the interface

File: src/Adapter/MSTest.TestAdapter/PlatformServiceProvider.cs [hunk 2]
```diff
@@ -150,5 +150,12 @@ public ITestSourceHost CreateTestSourceHost(
...
     /// </remarks>
-    public ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties) => new TestContextImplementation(testMethod, writer, properties);
+    public ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties)
+    {
+        var context = new TestContextImplementation(testMethod, writer, properties);
+        Token?.Register(() => context.CancellationTokenSource.Cancel());
+        return context;
+    }
+
+    public TestRunCancellationToken? Token { get; set; }
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/PlatformServiceProvider.cs]: Let's use full name here,

```suggestion
    public TestRunCancellationToken? TestRunCancellationToken { get; set; }
```

File: src/Adapter/MSTest.TestAdapter/PlatformServiceProvider.cs [hunk 3]
```diff
@@ -150,5 +150,12 @@ public ITestSourceHost CreateTestSourceHost(
     /// <remarks>
     /// This was required for compatibility reasons since the TestContext object that the V1 adapter had for desktop is not .Net Core compliant.
     /// </remarks>
-    public ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties) => new TestContextImplementation(testMethod, writer, properties);
+    public ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties)
+    {
+        var context = new TestContextImplementation(testMethod, writer, properties);
+        Token?.Register(() => context.CancellationTokenSource.Cancel());
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/PlatformServiceProvider.cs]: We can simplify the expression

```suggestion
        Token?.Register(context.CancellationTokenSource.Cancel);
```

File: src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@ internal FastFilter(ImmutableDictionary<string, ISet<string>> filterProperties,
     internal string? PropertyValueRegexReplacement { get; set; }
 
     internal string[]? ValidForProperties(IEnumerable<string>? properties) => properties is null
-            ? (string[]?)null
+            ? null
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs]: Let's try to keep PRs focused on the topic, this deserve a different PR (I did it before this change).

File: src/Adapter/MSTest.TestAdapter/IPlatformServiceProvider.cs (hunk 1 above)
Comment (engyebrahim) [reply src/Adapter/MSTest.TestAdapter/IPlatformServiceProvider.cs]: without changing it and changing the return type of the instance that makes a lot of tests fail

Comment (Evangelink) [reply src/Adapter/MSTest.TestAdapter/IPlatformServiceProvider.cs]: I have tried and indeed many tests are replacing the instance with a testable one that only implements the interface.

File: src/Adapter/MSTest.TestAdapter/PlatformServiceProvider.cs [hunk 5]
```diff
@@ -150,5 +150,12 @@ public ITestSourceHost CreateTestSourceHost(
...
     /// </remarks>
-    public ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties) => new TestContextImplementation(testMethod, writer, properties);
+    public ITestContext GetTestContext(ITestMethod testMethod, StringWriter writer, IDictionary<string, object?> properties)
+    {
+        var context = new TestContextImplementation(testMethod, writer, properties);
+        TestRunCancellationToken?.Register(context.CancellationTokenSource.Cancel);
+        return context;
+    }
+
+    public TestRunCancellationToken? TestRunCancellationToken { get; set; }
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/PlatformServiceProvider.cs]: Let's try to keep properties together and let's add a comment on top of that property to explain what he does and why it's needed.

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AbortionTests.cs [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+using System.Runtime.InteropServices;
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+using Microsoft.Testing.Platform.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestGroup]
+public class AbortionTests : AcceptanceTestBase
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AbortionTests.cs]: Nit:

```suggestion
public sealed class AbortionTests : AcceptanceTestBase
```

**Review by Evangelink — COMMENTED** [review]
Will take your branch and do some manual tests and checks

**Review by MarcoRossignoli — APPROVED** [review]
